pkg/dataobj/sections/streams/decoder.go

```diff
@@ -162,6 +174,8 @@ func (rd *decoder) ReadPages(ctx context.Context, pages []*streamsmd.PageDesc) r
 			bufpool.Put(buffer)
 		}
 
+		level.Debug(utillog.WithContext(ctx, utillog.Logger)).Log("msg", "streams.decoder: read pages", "num_pages", len(pages), "window_size", humanize.Bytes(windowing.S3WindowSize), "total_windows", numWindows)
```

Comment:
Would it be useful to add the requested bytes here too? Up to you - I don't know the intent behind the log line

Reply:
this is added to track how many windows/requests we end up making when downloading a batch of pages. helps tune the window_size to reduce the number of roundtrips.